### PR TITLE
Backend/matching

### PR DESCRIPTION
**Pull request by 1mozolacal**
-All user info returned at the '/matching/:id' endpoint.
-fixed bug of liking someone you already matched with

### CODE REVIEW

File: duos-api/src/app.module.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { PreauthMiddleware } from './auth/preauth.middleware';
 export class AppModule {
   configure(consumer: MiddlewareConsumer) {
     consumer.apply(PreauthMiddleware).forRoutes({
-      path: '*', method: RequestMethod.ALL
+      path: 'ignore/*', method: RequestMethod.ALL
```
Comment (Vallyath) [inline duos-api/src/app.module.ts]: why was this changed?

File: duos-api/src/matching/matching.service.ts [hunk 2]
```diff
@@ -36,7 +47,7 @@ export class MatchingService {
                 await addMatch(likerId, likedId)
                 return {
                     matched: true,
-                    matchedId: likedId
+                    matchedId: likedUser
```
Comment (Vallyath) [inline duos-api/src/matching/matching.service.ts:54]: not sure if this is relevant but you can just put likedUser.uid here since I don't think the frontend will be doing anything with the matches besides showing a pop-up @dannykhuu2 correct me if I'm wrong here

**Review by Vallyath — CHANGES_REQUESTED** [review]
I think it's all good besides the two things mentioned

File: duos-api/src/matching/matching.service.ts (hunk 2 above)
Comment (dannykhuu2) [reply duos-api/src/matching/matching.service.ts:54]: Yea all im going to do is show a pop-up saying matched.

File: duos-api/src/app.module.ts (hunk 1 above)
Comment (1mozolacal) [reply duos-api/src/app.module.ts]: that was for testing, thanks for catching that

File: duos-api/src/matching/matching.service.ts (hunk 2 above)
Comment (1mozolacal) [reply duos-api/src/matching/matching.service.ts:54]: I am provided extra info if needed

Comment (1mozolacal) [reply duos-api/src/matching/matching.service.ts:54]: It can be changed to whatever @dannykhuu2 wants

Comment (Vallyath) [reply duos-api/src/matching/matching.service.ts:54]: it's fine you can just keep it as is then, not that it'll really matter

Comment (dannykhuu2) [reply duos-api/src/matching/matching.service.ts:54]: I think we should change it to what David suggested

**Review by Vallyath — APPROVED** [review]
looks good to me
